### PR TITLE
Added code for test TEST-40440, TEST-40441, TEST-40442, TEST-40443

### PR DESCRIPTION
**Pull request by pragamj**
Signed-off-by: pragam <pragam.jain@seagate.com>

# Problem Statement
-F-87K tests Automation 

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
[TEST-40440_TestNamespaceDeployment_20220427033257.log](https://github.com/Seagate/cortx-test/files/8581763/TEST-40440_TestNamespaceDeployment_20220427033257.log)
[TEST-40441_TestNamespaceDeployment_20220427232314.log](https://github.com/Seagate/cortx-test/files/8581764/TEST-40441_TestNamespaceDeployment_20220427232314.log)
[TEST-40443_TestNamespaceDeployment_20220428032913.log](https://github.com/Seagate/cortx-test/files/8581765/TEST-40443_TestNamespaceDeployment_20220428032913.log)

-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)
[collections_log.txt](https://github.com/Seagate/cortx-test/files/8581773/collections_log.txt)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by pragamj** [comment]
TEST-40442 is blocked due to CORTX-31052

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 1]
```diff
@@ -1728,3 +1746,52 @@ def generate_and_compare_both_version(input_installing_version, installed_versio
...
+        master_node_list.execute_cmd(common_cmd.KUBECTL_DEL_NAMESPACE.format(namespace),
+                                     read_lines=True)
+        resp = master_node_list.execute_cmd(common_cmd.KUBECTL_GET_NAMESPACE, read_lines=True)
+        LOGGER.debug("The namespace is %s", resp.pop(0))
+        if namespace in resp:
+            return False, f"Failed to del namespace {namespace}"
+        return True, f"Successfully deleted the NAMESPACE {namespace}"
+
+    @staticmethod
+    def aphanumeric_string_generator(size):
```
Comment (ketanarlulkar) [inline libs/prov/prov_k8s_cortx_deploy.py:1785]: We have it already. Check `random_string_generator` from `commons/utils/system_utils.py`

Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py:1785]: those were either having lowercase or upper and lower case mix , we explicit need digits and lowercase only , uppercase is not supported.

Comment (ketanarlulkar) [reply libs/prov/prov_k8s_cortx_deploy.py:1785]: you can pass `chars=string.digits + string.ascii_lowercase` parameter to that function.

**Review by pranav-sahasrabudhe — APPROVED** [review]
Approved

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 2]
```diff
@@ -1728,3 +1746,52 @@ def generate_and_compare_both_version(input_installing_version, installed_versio
...
+        This method generate random string with combination of lowercase,digit,`-`
+        and returns aphanumeric string
+        param: size : length of string
+        returns Alphanumeric String with `-`
+        """
+        char = string.ascii_lowercase + string.digits
+        generated_string = ''.join(secrets.choice(char) for _ in range(size))
+        string_len = int(len(generated_string) / 2)
+        string_alpha = generated_string[:string_len] + "-" + generated_string[string_len:]
+        LOGGER.info("The string is %s", string_alpha)
```
Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:1796]: Print length of string as well.

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 3]
```diff
@@ -1270,19 +1282,25 @@ def test_deployment(self, sns_data, sns_parity,
...
-            LOGGER.info("service resp is %s", service_status)
-            assert_utils.assert_true(service_status[0], service_status[1])
-            row.append(service_status[-1])
-            if self.deployment_type != self.deploy_cfg["deployment_type_data"]:
-                if self.cortx_server_image:
-                    resp = self.verfiy_installed_rpms(master_node_list,
-                                                      common_const.RGW_CONTAINER_NAME,
-                                                      self.deploy_cfg["rgw_rpm"])
-                    assert_utils.assert_true(resp[0], resp[1])
+            if len(namespace) >= 63 or bool(re.match(r'\w*[A-Z]\w*', namespace)):
```
Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:1285]: add 63 to config.

Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:1285]: For positive scenario (continuous deployment), validate if length < 63 and all character are lower case.

**Review by priyankaborawake — APPROVED** [review]
Added few minor comments, 
LGTM. Approved

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 4]
```diff
@@ -1270,19 +1282,25 @@ def test_deployment(self, sns_data, sns_parity,
...
+                LOGGER.debug("Negative Test Scenario")
+                assert_utils.assert_false(resp[0], resp[1])
+            else:
+                assert_utils.assert_true(resp[0], resp[1])
+            matches = [re.compile(pat) for pat in self.patterns]
+            if not (m.match(resp[1]) for m in matches):
+                LOGGER.info("Step to Check  ALL service status")
+                time.sleep(60)
```
Comment (sonalk0209) [inline libs/prov/prov_k8s_cortx_deploy.py:1293]: can we add polling rather?

**Review by sonalk0209 — COMMENTED** [review]
@pragamj few comments

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 5]
```diff
@@ -84,6 +86,7 @@ def __init__(self):
         self.data_only_list = ["data-only", "standard"]
         self.server_only_list = ["server-only", "standard"]
         self.exclusive_pod_list = ["data-only", "server-pod"]
+        self.patterns = ['RFC 1123', '63 characters']
```
[the remark this answers is not in the document]
Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py:89]: these are patterns which we receive in error message for invalid namespace name while running deploy script. we have just matched with part of the string in error message.

File: libs/prov/prov_k8s_cortx_deploy.py (hunk 3 above)
Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py]: done

Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py]: done

**Review by ketanarlulkar — APPROVED** [review]

**Review by sonalk0209 — APPROVED** [review]
Approving for merge.

**Pull request merged**